libbeat/cmd/instance/beat_test.go

```diff
@@ -433,3 +433,16 @@ output:
 		})
 	}
 }
+
+func TestBeat_initConfigManager(t *testing.T) {
```

Comment:
I'm not happy with that, but without a huge refactor, I did not see another way to test it. An integration test in the agent seems to me the best approach. However I did not want to let it here completely untested

Reply:
I see the reasoning but I don't think this test is really testing anything. All it is testing is that if you call `initConfigManager(true)` that `b.Info.Version` is empty. 

It doesn't test any part of the `initConfigManager(false)` path that we actually care about. Allowing the ability to override the NewManager call with a different implementation of the interface that does the same thing isn't super valuable for the same reason, and it is more code.

Tiago added integration tests that run the Beat independently but give control over the agent control protocol messaging, that might be a better way to test this:

See https://github.com/elastic/beats/blob/0c73852e6429f68fd6a897a30e734d7deff940a3/x-pack/filebeat/tests/integration/managerV2_test.go#L45-L53

Reply:
Part of the problem is the version you want to verify as overrided is primarily going to show up in the actual published events, so the best test is to use the agent testing framework to actually query those resulting documents when Beats are running under agent and make sure they report what they should.